### PR TITLE
DVEO ASI sink

### PR DESCRIPTION
**Pull request by funman**
This is an RFC.

Currently missing: 

- latency handling
- scope testing to make sure output rate is correct after dejittering input stream
- maybe options handling (buffers/bufsize)
- maybe another way to open the card (wait until first packet is received?)

Comments welcome!

### CODE REVIEW

File: lib/upipe-dveo/upipe_dveo_asi_sink.c [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+
+    /* write header */
+    int size = 8;
+    uint8_t *header_write_ptr;
+    if (!ubase_check(ubuf_block_write(header, 0, &size, &header_write_ptr))) {
+        upipe_err(upipe, "could not write header");
+        ubuf_free(header);
+        return true;
+    }
+    memcpy(header_write_ptr, timestamp.timestamp, 8);
```
Comment (cmassiot) [inline lib/upipe-dveo/upipe_dveo_asi_sink.c]: Are you sure the timestamp is in the native endianness ? If so, can you add a comment saying you checked ?

File: lib/upipe-dveo/upipe_dveo_asi_sink.c [hunk 2]
```diff
@@ -0,0 +1,765 @@
...
+    }
+
+    unsigned long int mode;
+    snprintf(sys, sizeof(sys), sys_fmt, upipe_dveo_asi_sink->card_idx, "mode");
+    if (util_strtoul(sys, &mode) < 0) {
+        upipe_err(upipe, "Couldn't read buffer size");
+        goto error;
+    }
+
+    unsigned int packetsize;
```
Comment (cmassiot) [inline lib/upipe-dveo/upipe_dveo_asi_sink.c]: Dead code.

Comment (funman) [reply lib/upipe-dveo/upipe_dveo_asi_sink.c]: Which code is dead? github web UI and email do not show the same code and I'm not sure what's wrong here.

Comment (cmassiot) [reply lib/upipe-dveo/upipe_dveo_asi_sink.c]: The packetsize calculation doesn't seem used anywhere.

**Comment by cmassiot** [comment]
LGTM

**Comment by funman** [comment]
Rebased all commits into one

**Pull request merged**